### ISSUE TITLE
OML Plugins / Feature Fails to Install / Update via Eclipse "Install New Software..."

### ISSUE THREAD

**Issue by sjiherzig** [JPL-IMCE/gov.nasa.jpl.imce.oml#73]
When downloading a "vanilla" Eclipse + Modeling, simply installing the OML plugins via the "Install New Software..." mechanism (where I point to the update site) detects the relevant OML plugin versions, but fails with "Cannot complete action". Eclipse tries to compute alternative solutions, but none of them permit installing OML due to dependencies being broken (a long list of broken dependencies to internal OML plugins appears with various versions - some of which are "0.0"). This leads me to believe that there is something wrong with the ...updatesite project.

This behavior can be easily reproduced by starting with a fresh install of Eclipse (Neon 3.x in my case).

**Comment by NicolasRouquette** [comment]
The OML WB depends on Eclipse Neon.3, Xtext 2.11 and several features from Obeo's designer.
All of these dependencies are defined in one place:

https://github.com/JPL-IMCE/gov.nasa.jpl.imce.oml.tycho/blob/master/releng/gov.nasa.jpl.imce.oml.target/gov.nasa.jpl.imce.oml.target.target

The OML UpdateSite is small but it depends on the above target.

So, to install the OML update site features, it is necessary to have the above target first.

I suggest trying to get the above target definition and install that.
Then, the OML update site should install cleanly in that target.

**Comment by sjiherzig** [comment]
I did have most of these (Xtext 2.11, Neon 3) except for the Obeo designer components). It would be good to add these as dependencies in the update site configuration as well - that way, any missing features will be displayed in the "Install New Software..." dialog.

**Comment by sjiherzig** [comment]
BTW - this also fails when trying to update from 0.6.0 to 0.7.1.2 - i.e., within an OML workbench instance that worked fine. Not sure how drastic the changes are from 0.6.0 to 0.7.1.2 - did Obeo only make it into this latest release?

**Comment by NicolasRouquette** [comment]
FYI: there are two different update sites:
gov.nasa.jpl.imce.oml.platform.updatesite
gov.nasa.jpl.imce.oml.updatesite

The first defines the target platform for building OML;
the second defines the features that constitute the OML WB.

Have you tried to install gov.nasa.jpl.imce.oml.platform.updatesite first, and then gov.nasa.jpl.imce.oml.updatesite?

**Comment by sjiherzig** [comment]
No, only from gov.nasa.jpl.imce.oml.updatesite ... what is the proper procedure? Could you document this, please?

**Comment by NicolasRouquette** [comment]
Part of the simplification of the eclipse/tycho build in #75 will include documenting the process for updating the OML WB.

**Comment by castillr** [comment]
I will be investigating creation of a separate gov.nasa.jpl.imce.oml.p2 repository on bintray. Such that we can have a clean decoupling between maven and p2 repositories.

**Comment by castillr** [comment]
I haven't looked into this any further.

**Comment by NicolasRouquette** [comment]
Tested the following:

1) Install Eclipse Neon.3 package for Java EE Developers
https://www.eclipse.org/downloads/packages/eclipse-ide-java-ee-developers/neon3

2) In Eclipse, add update sites for:
Xtext: http://download.eclipse.org/modeling/tmf/xtext/updates/composite/releases/
OML WB: https://dl.bintray.com/jpl-imce/gov.nasa.jpl.imce.oml/

3) Install new features: Select just "OML WB" 
(the current version is 0.8.0.0-M26)

This will install Xtext 2.12 (not 2.11) and the OML WB.

**Comment by castillr** [comment]
Does this also work for separate features of the OML WB? For example, i will likely not want to install the entire WB, but only oml.model

**Comment by NicolasRouquette** [comment]
To make this work, I used the Jfrog CLI to "explode" the update site zip:

```
$ jfrog bt u --explode true ~/Downloads/gov.nasa.jpl.imce.oml.updatesite-0.8.0.0-M26.zip jpl-imce/gov.nasa.jpl.imce.oml/gov.nasa.jpl.imce.oml.updatesite/0.8.0.0-M26 
[Info] Verifying repository gov.nasa.jpl.imce.oml exists...
[Info] Verifying package gov.nasa.jpl.imce.oml.updatesite exists...
[Info] Verifying version 0.8.0.0-M26 exists...
[Info] Creating version...
[Info] Created version 0.8.0.0-M26.
[Info] [Thread 0] Uploading artifact: /Users/rouquett/Downloads/gov.nasa.jpl.imce.oml.updatesite-0.8.0.0-M26.zip
[Info] Uploaded 1 artifacts.
$ jfrog bt vp jpl-imce/gov.nasa.jpl.imce.oml/gov.nasa.jpl.imce.oml.updatesite/0.8.0.0-M26 
[Info] Publishing version...
[Info] Published version 0.8.0.0-M26, details:
{
  "files": 8
}
```
